parsl/dataflow/dflow.py

```diff
@@ -1158,15 +1158,11 @@ def checkpoint(self, tasks=None):
                         t = {'hash': hashsum,
                              'exception': None,
                              'result': None}
-                        try:
-                            # Asking for the result will raise an exception if
-                            # the app had failed. Should we even checkpoint these?
-                            # TODO : Resolve this question ?
-                            r = self.memoizer.hash_lookup(hashsum).result()
-                        except Exception as e:
-                            t['exception'] = e
-                        else:
-                            t['result'] = r
+
+                        fut = self.memoizer.hash_lookup(hashsum)
+                        assert fut.done()
+                        assert fut.exception() is None
```

Comment:
Aren't we already checking this in the condition above on line 1152?

Reply:
The test on line 1152 checks  self.tasks[task_id]['app_fu'], not self.memoizer.hash_lookup(hashsum)

Is the result of the hash_lookup meant to be the same as self.tasks[task_id]['app_fu'] ? I think probably so?
 
In which case that hash_lookup call is probably unnecessary?

Reply:
(i.e. the task record should be the source of truth here, not the hash lookup table?)